docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.module

```diff
@@ -58,3 +58,56 @@ function hs_paragraph_types_field_widget_complete_paragraphs_form_alter(array &$
     }
   }
 }
+
+/**
+ * Implements hook_field_widget_complete_WIDGET_TYPE_form_alter().
+ */
+function hs_paragraph_types_field_widget_complete_paragraphs_browser_form_alter(array &$field_widget_form, FormStateInterface $form_state, array $context) {
+  // There are multiple issues with setting and removing the required state
+  // from paragraph fields when using the `conditional_fields` module, so
+  // use form alter to use the states api instead.
+  // @see https://www.drupal.org/project/conditional_fields/issues/3344587
+  // Check that hs_collection is the correct paragraph type
+  // and correct fields are present.
+  if (!empty($field_widget_form['widget']['#field_name']) && $field_widget_form['widget']['#field_name'] == 'field_hs_page_components') {
```

Comment:
so many nested conditions. use escape conditions to make things more easily readable.

Reply:
Fixed.